.ci/Jenkinsfile

```diff
@@ -105,6 +105,22 @@ pipeline {
       }
     }
 
+    stage('E2e Test') {
+      steps {
+        withGithubNotify(context: 'E2e Test') {
+          cleanup()
+          withNodeEnv(){
+            dir("${BASE_DIR}"){
+              sh(label: 'install dependencies',script: 'npm i')
```

Comment:
npm run test already installs the packages. May be we should remove the install script from one place.

Reply:
it doesn't install root deps actually.

Reply:
locally it fails without this step, so I guess it will fail on CI too.

```
git clone https://github.com/elastic/synthetics.git
cd synthetics/__tests__/e2e
npm run test
```

```
> git clone https://github.com/elastic/synthetics.git
Cloning into 'synthetics'...
remote: Enumerating objects: 133, done.
remote: Counting objects: 100% (133/133), done.
remote: Compressing objects: 100% (92/92), done.
remote: Total 1442 (delta 69), reused 77 (delta 40), pack-reused 1309
Receiving objects: 100% (1442/1442), 643.39 KiB | 1.04 MiB/s, done.
Resolving deltas: 100% (853/853), done.
~/src/temp took 2s 
> cd synthetics/__tests__/e2e
synthetics/__tests__/e2e on  master via ⬢ v13.12.0 
> npm run test

> e2e-synthetics@ test /Users/inifc/src/temp/synthetics/__tests__/e2e
> npm i && npm run setup &&  bash scripts/test.sh

npm WARN rollback Rolling back @types/node@14.14.10 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/@types/node is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm WARN rollback Rolling back @types/yauzl@2.9.1 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/@types/yauzl is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm WARN rollback Rolling back arg@4.1.3 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/arg is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm WARN rollback Rolling back atomic-sleep@1.0.0 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/atomic-sleep is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm WARN rollback Rolling back balanced-match@1.0.0 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/balanced-match is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm WARN rollback Rolling back buffer-crc32@0.2.13 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/buffer-crc32 is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm WARN rollback Rolling back buffer-from@1.1.1 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/buffer-from is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm WARN rollback Rolling back commander@6.2.0 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/commander is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm WARN rollback Rolling back concat-map@0.0.1 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/concat-map is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm WARN rollback Rolling back brace-expansion@1.1.11 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/brace-expansion is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm WARN rollback Rolling back diff@4.0.2 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/diff is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm WARN rollback Rolling back flatstr@1.0.12 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/flatstr is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm WARN rollback Rolling back fs.realpath@1.0.0 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/fs.realpath is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm WARN rollback Rolling back graceful-fs@4.2.4 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/graceful-fs is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm WARN rollback Rolling back inherits@2.0.4 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/inherits is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm WARN rollback Rolling back jpeg-js@0.4.2 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/jpeg-js is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm WARN rollback Rolling back kleur@4.1.3 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/kleur is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm WARN rollback Rolling back make-error@1.3.6 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/make-error is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm WARN rollback Rolling back map-obj@4.1.0 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/map-obj is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm WARN rollback Rolling back mime@2.4.6 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/mime is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm WARN rollback Rolling back minimatch@3.0.4 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/minimatch is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm WARN rollback Rolling back ms@2.1.2 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/ms is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm WARN rollback Rolling back debug@4.3.1 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/debug is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm WARN rollback Rolling back agent-base@6.0.2 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/agent-base is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm WARN rollback Rolling back https-proxy-agent@5.0.0 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/https-proxy-agent is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm WARN rollback Rolling back path-is-absolute@1.0.1 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/path-is-absolute is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm WARN rollback Rolling back pend@1.2.0 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/pend is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm WARN rollback Rolling back fd-slicer@1.1.0 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/fd-slicer is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm WARN rollback Rolling back pngjs@5.0.0 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/pngjs is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm WARN rollback Rolling back progress@2.0.3 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/progress is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm WARN rollback Rolling back proxy-from-env@1.1.0 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/proxy-from-env is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm WARN rollback Rolling back retry@0.12.0 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/retry is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm WARN rollback Rolling back signal-exit@3.0.3 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/signal-exit is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm WARN rollback Rolling back proper-lockfile@4.1.1 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/proper-lockfile is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm WARN rollback Rolling back sonic-boom@1.3.0 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/sonic-boom is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm WARN rollback Rolling back source-map@0.6.1 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/source-map is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm WARN rollback Rolling back source-map-support@0.5.19 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/source-map-support is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm WARN rollback Rolling back to-no-case@1.0.2 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/to-no-case is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm WARN rollback Rolling back to-space-case@1.0.0 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/to-space-case is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm WARN rollback Rolling back to-snake-case@1.0.0 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/to-snake-case is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm WARN rollback Rolling back snakecase-keys@3.2.0 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/snakecase-keys is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm WARN rollback Rolling back totalist@2.0.0 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/totalist is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm WARN rollback Rolling back typescript@3.9.7 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/typescript is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm WARN rollback Rolling back wrappy@1.0.2 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/wrappy is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm WARN rollback Rolling back once@1.4.0 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/once is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm WARN rollback Rolling back end-of-stream@1.4.4 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/end-of-stream is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm WARN rollback Rolling back inflight@1.0.6 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/inflight is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm WARN rollback Rolling back glob@7.1.6 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/glob is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm WARN rollback Rolling back rimraf@3.0.2 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/rimraf is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm WARN rollback Rolling back pump@3.0.0 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/pump is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm WARN rollback Rolling back get-stream@5.2.0 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/get-stream is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm WARN rollback Rolling back ws@7.4.0 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/ws is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm WARN rollback Rolling back yauzl@2.10.0 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/yauzl is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm WARN rollback Rolling back extract-zip@2.0.1 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/extract-zip is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm WARN rollback Rolling back playwright-chromium@1.6.2 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/playwright-chromium is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm WARN rollback Rolling back yn@3.1.1 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/yn is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm WARN rollback Rolling back ts-node@8.10.2 failed (this is probably harmless): /Users/inifc/src/temp/synthetics/node_modules/ts-node is not a child of /Users/inifc/src/temp/synthetics/__tests__/e2e
npm ERR! code ENOENT
npm ERR! syscall chmod
npm ERR! path /Users/inifc/src/temp/synthetics/__tests__/e2e/node_modules/@elastic/synthetics/dist/cli.js
npm ERR! errno -2
npm ERR! enoent ENOENT: no such file or directory, chmod '/Users/inifc/src/temp/synthetics/__tests__/e2e/node_modules/@elastic/synthetics/dist/cli.js'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent 

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/inifc/.npm/_logs/2020-11-25T16_04_58_325Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 254
npm ERR! e2e-synthetics@ test: `npm i && npm run setup &&  bash scripts/test.sh`
npm ERR! Exit status 254
npm ERR! 
npm ERR! Failed at the e2e-synthetics@ test script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/inifc/.npm/_logs/2020-11-25T16_04_58_353Z-debug.log
```

Reply:
```
git clone https://github.com/elastic/synthetics.git
cd synthetics
npm i
cd __tests__/e2e
npm run test
```

```
 git clone https://github.com/elastic/synthetics.git
Cloning into 'synthetics'...
remote: Enumerating objects: 133, done.
remote: Counting objects: 100% (133/133), done.
remote: Compressing objects: 100% (92/92), done.
remote: Total 1442 (delta 69), reused 77 (delta 40), pack-reused 1309
Receiving objects: 100% (1442/1442), 643.39 KiB | 342.00 KiB/s, done.
Resolving deltas: 100% (853/853), done.
~/src/temp took 4s 
> cd synthetics
synthetics on  master via ⬢ v13.12.0 
> npm i
cd __tests__/e2e

> husky@4.3.0 install /Users/inifc/src/temp/synthetics/node_modules/husky
> node husky install

husky > Setting up git hooks
husky > Done

> playwright-chromium@1.6.2 install /Users/inifc/src/temp/synthetics/node_modules/playwright-chromium
> node install.js

Failed to process descriptor at /Users/inifc/.npm/_npx/58879/lib/node_modules/@elastic/synthetics/node_modules/playwright-chromium

> husky@4.3.0 postinstall /Users/inifc/src/temp/synthetics/node_modules/husky
> opencollective-postinstall || exit 0

Thank you for using husky!
If you rely on this package, please consider supporting our open collective:
> https://opencollective.com/husky/donate


> nodemon@2.0.4 postinstall /Users/inifc/src/temp/synthetics/node_modules/nodemon
> node bin/postinstall || exit 0

Love nodemon? You can now support the project via the open collective:
 > https://opencollective.com/nodemon/donate

npm WARN prepublish-on-install As of npm@5, `prepublish` scripts are deprecated.
npm WARN prepublish-on-install Use `prepare` for build steps and `prepublishOnly` for upload-only.
npm WARN prepublish-on-install See the deprecation note in `npm help scripts` for more information.

> @elastic/synthetics@0.0.1-alpha.6 prepublish /Users/inifc/src/temp/synthetics
> npm run clean && npm run build


> @elastic/synthetics@0.0.1-alpha.6 clean /Users/inifc/src/temp/synthetics
> rm -rf dist


> @elastic/synthetics@0.0.1-alpha.6 build /Users/inifc/src/temp/synthetics
> tsc

> cd __tests__/e2e

synthetics/__tests__/e2e on  master via ⬢ v13.12.0 
> npm run test

> e2e-synthetics@ test /Users/inifc/src/temp/synthetics/__tests__/e2e
> npm i && npm run setup &&  bash scripts/test.sh

added 3 packages from 4 contributors and audited 60 packages in 2.18s

44 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities


> e2e-synthetics@ setup /Users/inifc/src/temp/synthetics/__tests__/e2e
> bash scripts/setup.sh


Temporary folder
Temporary files will be stored in: ./tmp

Starting elasticsearch , kibana and synthetics docker


Waiting for synthetics docker to start...
Error: No such object: synthetics

...

✅ Setup completed successfully. Running e2e tests...
Waiting for Elastic Search  to start
Waiting for Synthetics to send data to ES for test monitor

Journey: E2e test synthetics
   ✓  Step: 'Go to kibana uptime app' succeeded (3075 ms)
   ✓  Step: 'Check if there is table data' succeeded (9436 ms)
   ✓  Step: 'Click on my monitor' succeeded (1315 ms)
   ✓  Step: 'It navigates to details page' succeeded (68 ms)

 4 passed (296350 ms) 

```

Reply:
so `npm i` at top level is needed.

Reply:
actually, in the pipeline, I can rid of it because we make the `npm install` at the beginning and we stash the sources and dependencies to use them across the stages.